vision/google/cloud/vision/image.py

```diff
@@ -85,28 +85,58 @@ def source(self):
         """
         return self._source
 
-    def _detect_annotation(self, feature):
+    def _detect_annotation(self, features):
         """Generic method for detecting a single annotation.
 
-        :type feature: :class:`~google.cloud.vision.feature.Feature`
-        :param feature: The ``Feature`` indication the type of annotation to
-                        perform.
+        :type features: list of:class:`~google.cloud.vision.feature.Feature`
+        :param features: List of the ``Feature``s indicating the type of
+                         annotations to perform.
 
         :rtype: list
         :returns: List of
-                  :class:`~google.cloud.vision.entity.EntityAnnotation`.
+                  :class:`~google.cloud.vision.entity.EntityAnnotation`,
+                  :class:`~google.cloud.vision.face.Face`,
+                  :class:`~google.cloud.vision.color.ImagePropertiesAnnotation`,
+                  :class:`~google.cloud.vision.sage.SafeSearchAnnotation`,
         """
+        detected_objects = []
+        results = self.client.annotate(self, features)
+        for feature in features:
+            detected_objects.extend(
+                self._entity_from_response_type(feature.feature_type, results))
+
+        if len(detected_objects) == 1:
+            return detected_objects[0]
+        return detected_objects
+
+    @staticmethod
+    def _entity_from_response_type(feature_type, results):
+        """Convert a JSON result to an entity type based on the feature."""
+
         reverse_types = {
+            'FACE_DETECTION': 'faceAnnotations',
```

Comment:
I realize this was already here, but please move this map to module-level scope and name it `_REVERSE_TYPES` (module-level mutables are scary, but it's non-public so if users break it they buy it)